### ISSUE TITLE
Error occurred during build process while linking floating point related instructions.

### ISSUE THREAD

**Issue by mingxien** [TF-Hafnium/hafnium#2]
Has anyone run into this build error?

```
[288/325] LINK secure_tc_clang/obj/test/vmapi/primary_with_secondaries/services/service_vm1.elf
FAILED: secure_tc_clang/obj/test/vmapi/primary_with_secondaries/services/service_vm1.elf
ld.lld -pie --gc-sections --defsym=ORIGIN_ADDRESS=0xfd000000 -O2 --icf=all --fatal-warnings --color-diagnostics -T hafnium/9999-r0/hafnium-9999/3rdparty/hafnium/build/image/image.ld -o secure_tc_clang/obj/test/vmapi/primary_with_secondaries/services/service_vm1.elf --start-group @secure_tc_clang/obj/test/vmapi/primary_with_secondaries/services/service_vm1.elf.rsp --end-group
ld.lld: error: ld-temp.o <inline asm>:1:2: instruction requires: fp-armv8
        fmov d0, x8
        ^


ld.lld: error: ld-temp.o <inline asm>:1:2: instruction requires: fp-armv8
        fmov d1, d0
        ^
```

The code was built using the recommended toolchain, which is clang+llvm-15.0.6-x86_64-linux-gnu-ubuntu-18.04.tar.xz.
I have tried to troubleshoot the issue, but it seems like the cflags `-mgeneral-regs-only` is not functioning expectedly.
Ref: https://issues.fuchsia.dev/issues/319847726

**Comment by odeprez** [comment]
Hi, thanks for reporting. Please ignore the bot message (I will take an action to mute it). Using github issues tracker is relatively new to us, but we actually actively monitor.

On the build failure, we've not faced this issue so far afaik, may I ask which host / distro you're using?
To be fair most of validation is done on Ubuntu 20 or 22. Using other kinds of distros might exhibit random issues.

**Comment by mingxien** [comment]
Hi @odeprez,

Yes, there are serveral changes on top of upstream. I took a closer look at the `build/toolchain/embedded.gni` file and realized that test VMs for `secure_tc` should have a postfix `_vm` as target so that it will use correct `aarch64_toolchain()` according to https://git.trustedfirmware.org/hafnium/hafnium.git/tree/build/toolchain/embedded.gni#n347.

The issue has been resolved, thanks for your response.
